src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndexByKey(key);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        while (currentNode.next != null
+                || currentNode.key == key || currentNode.key != null
+                && currentNode.key.equals(key)) {
+            if (currentNode.key == key || currentNode.key != null
+                    && currentNode.key.equals(key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        currentNode.next = new Node<>(key, value);
+        size++;
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndexByKey(key)];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
```

Comment:
in one place you're comparing keys with custom logic, here you use Objects class. be consistent and use only one approach